packages/next-on-pages/src/buildApplication/buildSummary.ts

```diff
@@ -0,0 +1,179 @@
+ * @param idx Current item index.
+ * @returns The prefix for the log message.
+ */
+function getItemPrefix(total: number, idx: number): string {
+	if (total === 1) return '-';
+	if (idx === 0) return '┌';
+	if (idx === total - 1) return '└';
+	return '├';
+}
+
+/**
+ * Adds a group of items to the build summary, limiting the number of items to be displayed if
+ * necessary.
+ *
+ * Formats the items with stylised prefixes and indentations.
+ *
+ * @param summary Current summary.
+ * @param name The name of the group of items.
+ * @param rawItems List of items to be added to the summary.
+ * @param limit Maximum number of items to be displayed.
+ * @returns The updated summary.
+ */
+function addToSummary(
+	summary: string,
+	name: string,
+	rawItems: string[],
+	limit?: number
+): string {
+	if (rawItems.length === 0) return summary;
+
+	const items =
+		limit && rawItems.length > limit
+			? [
+					...rawItems.slice(0, limit - 1),
+					`... ${rawItems.length - limit - 1} more`,
+			  ]
+			: rawItems;
+
+	const formattedItems = items
+		.map((path, idx) => `  ${getItemPrefix(items.length, idx)} ${path}`)
+		.join('\n');
+
+	return `${summary}
+	
+${name} (${rawItems.length})
+${formattedItems}`;
+}
+
+/**
+ * Prints a build summary to the console.
+ *
+ * @param staticAssets List of static assets collected during the build.
+ * @param processedVercelOutput Results of processing the Vercel output directory.
+ * @param directoryProcessingResults Results of processing the output directory.
+ */
+export function printBuildSummary(
+	staticAssets: string[],
+	{ vercelOutput }: ProcessedVercelOutput,
+	{
+		functionsMap = new Map(),
+		prerenderedRoutes = new Map(),
+		wasmIdentifiers = new Map(),
+	}: Partial<DirectoryProcessingResults> = {}
+): void {
+	const middlewareFunctions = [...vercelOutput.entries()]
+		.filter(([, { type }]) => type === 'middleware')
+		.map(([path]) => path)
+		.sort((a, b) => a.localeCompare(b));
+	const routeFunctions = new Set(
+		processItemsMap(functionsMap).filter(
+			path => !middlewareFunctions.includes(path.replace(/^\//, ''))
+		)
+	);
+	const prerendered = processItemsMap(prerenderedRoutes);
+	let otherStatic = staticAssets
+		.filter(path => !prerenderedRoutes.has(path))
+		.sort((a, b) => a.localeCompare(b));
+
+	// Move the `/_next` and `/__next` assets to the end of the static assets list
+	otherStatic = [
+		...otherStatic.filter(path => !/^\/__?next/.test(path)),
+		...otherStatic.filter(path => /^\/__?next/.test(path)),
+	];
+
+	let summary = `Build Summary (@cloudflare/next-on-pages v${nextOnPagesVersion})`;
+	summary = addToSummary(summary, 'Middleware Functions', middlewareFunctions);
+	summary = addToSummary(summary, 'Edge Function Routes', [...routeFunctions]);
+	summary = addToSummary(summary, 'Prerendered Routes', prerendered, 20);
+	summary = addToSummary(summary, 'Wasm Files', [...wasmIdentifiers.keys()]);
+	summary = addToSummary(summary, 'Other Static Assets', otherStatic, 5);
```

Comment:
what about something like this instead?
```suggestion
	const summaryTitle = `Build Summary (@cloudflare/next-on-pages v${nextOnPagesVersion})`;
	const summarySections = buildSummarySections([
		{ name: 'Middleware Functions', rawItems: middlewareFunctions },
		{ name: 'Edge Function Routes', rawItems: [...routeFunctions] },
		{ name: 'Prerendered Routes', rawItems: prerendered, limit: 20 },
		{ name: 'Wasm Files', rawItems: [...wasmIdentifiers.keys()] },
		{ name: 'Other Static Assets', rawItems: otherStatic, limit: 5 },
	]);
	const summary = `${summaryTitle}\n\n${summarySections}`;
```

Where `buildSummarySections` is something like:
```ts
function buildSummarySections(
	sections: {
		name: string;
		rawItems: string[];
		limit?: number;
	}[]
): string {
	return sections.map(({ name, rawItems, limit }) => {
		if (rawItems.length === 0) return null;

		const items =
			limit && rawItems.length > limit
				? [
						...rawItems.slice(0, limit - 1),
						`... ${rawItems.length - limit - 1} more`,
				  ]
				: rawItems;

		const formattedItems = items
			.map((path, idx) => `  ${getItemPrefix(items.length, idx)} ${path}`)
			.join('\n');

		return `${name} (${rawItems.length})\n${formattedItems}`;
	}).filter(Boolean).join('\n\n');
}
```

I think it could be a bit clearer and help us avoid `let`s with different iterations/modifications of the summary 🤔